### PR TITLE
Adding linter setup

### PR DESCRIPTION
**Pull request by hollyos**
Adding the linter setup from our core libraries to ensure consistency in our projects.

Also, pinned the package dependency versions and the node version (LTS) to ensure this package continues to function as expected going forward.

I added `Dev note`'s to the `index.ts` file to call out specific changes that are not pure linting

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -1,65 +1,77 @@
...
 
 function buildPath(startDate: string, endDate: string) {
-  return `/v2/packages/${scarfEntity}/events?start_date=${startDate}&end_date=${endDate}`
+  return `/v2/packages/${scarfEntity}/events?start_date=${startDate}&end_date=${endDate}`;
 }
 
 async function runCommand(command: string): Promise<string> {
   return new Promise((resolve, reject) => {
-    const process = spawn('bash', ['-c', command])
+    const process: ChildProcess = spawn("bash", ["-c", command]);
```
Comment (hollyos) [inline index.ts:23]: Updated type

File: index.ts [hunk 2]
```diff
@@ -1,65 +1,77 @@
...
-    let output = ''
-    let errorOutput = ''
+    let output = "";
+    let errorOutput = "";
 
-    process.stdout.on('data', (data: any) => {
-      output += data.toString()
-    })
+    if (process.stdout) {
+      process.stdout.on("data", (data: Buffer) => {
```
Comment (hollyos) [inline index.ts:29]: Dev Note: Updated type

File: index.ts [hunk 3]
```diff
@@ -1,65 +1,77 @@
...
+        output += data.toString();
+      });
+    }
 
-    process.stderr.on('data', (data: any) => {
-      errorOutput += data.toString()
-    })
+    if (process.stderr) {
+      process.stderr.on("data", (data: Buffer) => {
```
Comment (hollyos) [inline index.ts:35]: Dev Note: Updated type

File: index.ts [hunk 4]
```diff
@@ -1,65 +1,77 @@
...
+        errorOutput += data.toString();
+      });
+    }
 
-    process.on('close', (code: any) => {
+    process.on("close", (code: number) => {
```
Comment (hollyos) [inline index.ts:40]: Dev Note: Updated type

File: index.ts [hunk 5]
```diff
@@ -81,80 +93,85 @@ async function main() {
...
-
+      hostname: "api.scarf.sh",
+      path,
+      headers: {
+        Authorization: `Bearer ${scarfApiToken}`,
+        "Content-Type": "application/json",
+      },
+    };
+
+    http.get(options, function (response: IncomingMessage) {
```
Comment (hollyos) [inline index.ts:120]: Dev Note: Updated type

File: index.ts [hunk 6]
```diff
@@ -81,80 +93,85 @@ async function main() {
...
-        response.on('data', (chunk: any) => {
-          data += chunk
-        })
-        response.on('end', () => {
-          console.log(data)
-          reject('non 200 response from Scarf. Check your auth token')
-        })
+        console.log(`Response code: ${response.statusCode}`);
+        let data = "";
+        response.on("data", (chunk: Buffer) => {
```
Comment (hollyos) [inline index.ts:124]: Dev Note: Updated type

File: index.ts [hunk 7]
```diff
@@ -81,80 +93,85 @@ async function main() {
...
 
-      response.pipe(file)
+      response.pipe(file);
 
-      file.on('error', (err: any) => {
-        file.close()
-        console.error(`Error downloading CSV: ${err}`)
-        reject(err)
-      })
+      file.on("error", (err: NodeJS.ErrnoException) => {
```
Comment (hollyos) [inline index.ts:135]: Dev Note: Updated type

File: index.ts [hunk 8]
```diff
@@ -1,65 +1,77 @@
...
 
-    let output = ''
-    let errorOutput = ''
+    let output = "";
+    let errorOutput = "";
 
-    process.stdout.on('data', (data: any) => {
-      output += data.toString()
-    })
+    if (process.stdout) {
```
Comment (hollyos) [inline index.ts:28]: Dev Note: Added check to ensure `process.stdout` is defined

File: index.ts [hunk 9]
```diff
@@ -1,65 +1,77 @@
...
+      process.stdout.on("data", (data: Buffer) => {
+        output += data.toString();
+      });
+    }
 
-    process.stderr.on('data', (data: any) => {
-      errorOutput += data.toString()
-    })
+    if (process.stderr) {
```
Comment (hollyos) [inline index.ts:34]: Dev Note: Added check to ensure `process.stderr` is defined

File: package.json [hunk 10]
```diff
@@ -2,22 +2,36 @@
...
+    "buildAndRun": "npx tsc && node index.js",
+    "lint": "eslint . --ext .ts",
+    "lint:cache": "eslint . --ext .ts --cache",
+    "lint:fix": "eslint . --ext .ts --fix",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
   "dependencies": {
-    "@scarf/scarf": "^1.1.1",
-    "typescript": "^4.9.4"
+    "@scarf/scarf": "1.1.1",
```
Comment (hollyos) [inline package.json:20]: @aviaviavi should we bump the version here to `1.3.0`?

Comment (aviaviavi) [inline package.json]:
```suggestion
    "@scarf/scarf": "^1.3.0",
```

I'm not sure I'm sold on fully pinning dependencies that we have full control over. I'd rather users get updates we push out, as we need to test against breakages before we cut releases anyway. 

Also the latest stable version for @scarf/scarf is `1.3.0`.

**Review by aviaviavi — APPROVED** [review]
Looks good!

**Pull request merged**